src/luxonis_ml/data/dataset.py

```diff
@@ -397,162 +241,97 @@ def _init_path(self) -> None:
+            if os.path.splitext(file)[1] == ".parquet":
+                dfs.append(pd.read_parquet(os.path.join(self.annotations_path, file)))
+        if len(dfs):
+            return pd.concat(dfs)
+        else:
+            return None
 
-    def _log_time(self, note: Optional[str] = None, final: bool = False) -> None:
-        """Helper to log the time taken within a function to INFO"""
+    def _try_instance_id(
+        self, file: str, index: Optional[pd.DataFrame]
+    ) -> Optional[str]:
+        if index is None:
+            return None
 
-        if self.logger.isEnabledFor(logging.DEBUG):
-            if self.last_time is None:
-                self.last_time = time.time()
-            else:
-                new_time = time.time()
-                self.logger.debug(f"{note} took {(new_time-self.last_time)*1000} ms")
-                self.last_time = new_time
-                if final:
-                    self.last_time = None
-
-    def _get_sample_collection(self) -> Optional[str]:
-        """Gets the name of the mongo collection storing samples"""
-
-        res = list(self.conn.datasets.find({"_id": ObjectId(self.fo_dataset._doc.id)}))
-        if len(res):
-            return res[0]["sample_collection_name"]
+        if file in list(index["file"]):
+            matched = index[index["file"] == file]
+            if len(matched):
+                return list(matched["instance_id"])[0]
         else:
             return None
 
-    def _update_from_instance_ids(
-        self,
-        instance_ids: List[str],
-        attribute: str,
-        value: str,
-        main_only: bool = False,
-    ) -> List[ObjectId]:
-        """Updates an attribute of a group of samples given their instance IDs"""
-
-        try:
-            sample_collection = self._get_sample_collection()
-        except:
-            raise Exception("Cannot find sample collection name")
-
-        if not main_only:
-            samples = self.conn[sample_collection].find(
-                {"instance_id": {"$in": instance_ids}}
-            )
-            instance_ids = set()
-            for sample in samples:
-                gid = sample[self.source.name]["_id"]
-                group_instance_ids = list(
-                    self.conn[sample_collection].find(
-                        {f"{self.source.name}._id": gid}, {"instance_id": 1}
-                    )
-                )
-                for iid in group_instance_ids:
-                    instance_ids.add(iid["instance_id"])
-            instance_ids = list(instance_ids)
+    def _get_file_index(self) -> Optional[pd.DataFrame]:
+        index = None
+        if self.bucket_storage.value == "local":
+            file_index_path = os.path.join(self.metadata_path, "file_index.parquet")
+        else:
+            file_index_path = os.path.join(".luxonisml_tmp", "file_index.parquet")
+            try:
+                self.fs.get_file("metadata/file_index.parquet", file_index_path)
```

Comment:
So this would mean on public bucket we'd have a ".luxonisml_tmp" directory?

Reply:
No, it's a temporary local directory that gets removed after the `file_index.parquet` is uploaded to the bucket.